src/ring_theory/algebra_tower.lean

```diff
@@ -458,3 +458,49 @@ by exactI fg_of_injective (is_scalar_tower.to_alg_hom B₀ B C).to_linear_map
   (linear_map.ker_eq_bot.2 hBCi)
 
 end artin_tate
+
+section alg_hom_tower
+
+variables {A} {B} {C D : Type*} [comm_semiring A] [comm_semiring B] [comm_semiring C]
+[comm_semiring D] [algebra A B] [algebra B C] [algebra A C] [algebra A D] [is_scalar_tower A B C]
+
+/-- Restrict the domain of an alg_hom -/
+def alg_hom_restrict (f : C →ₐ[A] D) : B →ₐ[A] D := f.comp (is_scalar_tower.to_alg_hom A B C)
```

Comment:
```suggestion
def alg_hom.restrict (f : C →ₐ[A] D) : B →ₐ[A] D := f.comp (is_scalar_tower.to_alg_hom A B C)
```
would make this usable as `f.restrict`. Does it make sense to make `B` an explicit argument?

Reply:
Done. I also did this for extend_base (I hope that makes sense).